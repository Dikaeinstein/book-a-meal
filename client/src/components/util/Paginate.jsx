import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { inherits } from 'util';


class Paginate extends Component {
  constructor(props) {
    super(props);
    this.handleNext = this.handleNext.bind(this);
    this.handlePrevious = this.handlePrevious.bind(this);
  }

  handleNext(event) {
    event.preventDefault();
    this.props.onPageChange(this.props.nextUrl);
  }

  handlePrevious(event) {
    event.preventDefault();
    this.props.onPageChange(this.props.previousUrl);
  }

  render() {
    const btnStyle = {
      margin: '1rem .5rem',
      background: 'none',
      color: '#eb480b',
      border: 'none',
    };
    const paginationStyle = {
      display: 'flex',
      justifyContent: 'center',
      marginTop: '1rem',
    };
    const paginationTextStyle = { margin: '.25rem', background: 'inherits' };

    return (
      <div style={{ ...paginationStyle, ...this.props.style }}>
        {this.props.previousUrl &&
          <button
            onClick={this.handlePrevious}
            style={btnStyle}
          >
            <span style={paginationTextStyle}>Previous</span>
            <i className="fas fa-arrow-alt-circle-left" />
          </button>}
        {this.props.nextUrl &&
          <button
            onClick={this.handleNext}
            style={btnStyle}
          >
            <i className="fas fa-arrow-alt-circle-right" />
            <span style={paginationTextStyle}>Next</span>
          </button>}
      </div>
    );
  }
}

Paginate.defaultProps = {
  style: {},
  nextUrl: '',
  previousUrl: '',
};

Paginate.propTypes = {
  onPageChange: PropTypes.func.isRequired,
  nextUrl: PropTypes.string,
  previousUrl: PropTypes.string,
  style: PropTypes.objectOf(PropTypes.string),
};

export default Paginate;
