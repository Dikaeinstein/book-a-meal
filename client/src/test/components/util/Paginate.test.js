import React from 'react';
import { shallow } from 'enzyme';
import Paginate from '../../../components/util/Paginate';

describe('<Paginate />', () => {
  it('should match snapshot', () => {
    let tree = shallow(<Paginate onPageChange={jest.fn()} />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<Paginate
      onPageChange={jest.fn()}
      previousUrl="http://prevUrl"
    />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<Paginate
      onPageChange={jest.fn()}
      nextUrl="http://nextUrl"
    />);
    expect(tree).toMatchSnapshot();
  });
  it('should call handleNext when next is clicked', () => {
    const spy = jest.spyOn(Paginate.prototype, 'handleNext');
    const wrapper = shallow(<Paginate
      onPageChange={jest.fn()}
      nextUrl="http://nextUrl"
    />);
    wrapper.find('button').last()
      .simulate('click', { preventDefault: jest.fn() });
    expect(spy).toHaveBeenCalled();
  });
  it('should call handlePrevious when next is clicked', () => {
    const spy = jest.spyOn(Paginate.prototype, 'handlePrevious');
    const wrapper = shallow(<Paginate
      onPageChange={jest.fn()}
      previousUrl="http://prevUrl"
    />);
    wrapper.find('button').first()
      .simulate('click', { preventDefault: jest.fn() });
    expect(spy).toHaveBeenCalled();
  });
});
