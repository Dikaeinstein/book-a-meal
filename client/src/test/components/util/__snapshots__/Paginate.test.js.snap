// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginate /> should match snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
      "marginTop": "1rem",
    }
  }
/>
`;

exports[`<Paginate /> should match snapshot 2`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
      "marginTop": "1rem",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "#eb480b",
        "margin": "1rem .5rem",
      }
    }
  >
    <span
      style={
        Object {
          "background": "inherits",
          "margin": ".25rem",
        }
      }
    >
      Previous
    </span>
    <i
      className="fas fa-arrow-alt-circle-left"
    />
  </button>
</div>
`;

exports[`<Paginate /> should match snapshot 3`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
      "marginTop": "1rem",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "none",
        "border": "none",
        "color": "#eb480b",
        "margin": "1rem .5rem",
      }
    }
  >
    <i
      className="fas fa-arrow-alt-circle-right"
    />
    <span
      style={
        Object {
          "background": "inherits",
          "margin": ".25rem",
        }
      }
    >
      Next
    </span>
  </button>
</div>
`;
